lib/fika/compiler/type_checker.ex

```diff
@@ -429,6 +510,39 @@ defmodule Fika.Compiler.TypeChecker do
     end
   end
 
+  defp reset_scope_and_set_signature(env, {:function, _, {_name, args, _type, _}} = ast) do
+    current_signature = Fika.Compiler.TypeChecker.function_ast_signature(env[:module], ast)
+
+    env
+    |> Map.merge(%{
+      scope: %{},
+      has_effect: false,
+      current_signature: current_signature,
+      latest_called_function: if(env[:first_pass], do: env[:current_signature]),
```

Comment:
Do we set `first_pass` to true anywhere?

Reply:
It was being set to true as the default value for the struct. This might need a refactor now that the struct has been removed.

init_env still sets it, but most tests don't call it